piker/brokers/binance.py

```diff
@@ -141,16 +159,56 @@ def __init__(self) -> None:
         self._sesh.base_location = _url
         self._pairs: dict[str, Any] = {}
 
+        conf = get_config()
+        self.api_key = conf.get('api', {}).get('key')
+        self.api_secret = conf.get('api', {}).get('secret')
+
+        if self.api_key:
+            self._sesh.headers.update({'X-MBX-APIKEY': self.api_key})
+
+    def _get_signature(self, data: OrderedDict) -> str:
+        if not self.api_secret:
+            raise BrokerConfigurationError(
+                'Attempt to get a signature without setting up credentials'
+            )
+
+        query_str = '&'.join([
+            f'{_key}={value}'
+            for _key, value in data.items()])
+        logging.info(query_str)
+        msg_auth = hmac.new(
+            self.api_secret.encode('utf-8'),
+            query_str.encode('utf-8'),
+            hashlib.sha256
+        )
+        return msg_auth.hexdigest()
+
     async def _api(
         self,
         method: str,
-        params: dict,
+        params: Union[dict, OrderedDict],
+        signed: bool = False,
+        action: str = 'get'
     ) -> Dict[str, Any]:
-        resp = await self._sesh.get(
-            path=f'/api/v3/{method}',
-            params=params,
-            timeout=float('inf')
-        )
+
+        if signed:
+            params['signature'] = self._get_signature(params)
+
+        if action == 'get':
```

Comment:
@goodboy Here is a part I'm not confident in, you think I should just make separate methods for `get` / `post` or this is fine?

Reply:
Hmm good question.
This seems fine I think?

you could maybe even just proxy the method direct to `asks` -> `getattr(self._sesh, method)` ?

Not sure if that's too "dynamic" or wtv.

Reply:
Yeah love that option, always forget about `getattr`